pages/thank-you.jsx

```diff
@@ -1,27 +1,40 @@
-// this will ignore that Bad Invocation error which is a bug in JSXHint
-/* jshint -W067 */
-
 import React from 'react';
 import MozillaFooter from '../components/mozilla/footer.jsx';
 import Signup from '../components/signup.jsx';
 import Social from '../components/social.jsx';
 import ThankYouHeader from '../components/thank-you-header.jsx';
-import { IntlMixin } from 'react-intl';
 import analytics from '../assets/js/analytics.js';
 import form from '../scripts/form.js';
+import qs from 'qs';
 
 var ThankYou = React.createClass({
-  mixins: [IntlMixin],
+  contextTypes: {
+    location: React.PropTypes.object,
+    router: React.PropTypes.object,
+    intl: React.PropTypes.object
+  },
+  propTypes: {
+    country: React.PropTypes.string.isRequired,
+    test: React.PropTypes.string,
+    location: React.PropTypes.object
+  },
+  getInitialState: function() {
+    let query = qs.parse(this.props.location.search.replace("?", ""));
```

Comment:
The location is in contextType
On Apr 3, 2016 6:49 PM, "ScottDowne" <notifications@github.com> wrote:

> In pages/thank-you.jsx
> <https://github.com/mozilla/donate.mozilla.org/pull/1444#discussion_r58319204>
> :
>
> >
> >  var ThankYou = React.createClass({
> > -  mixins: [IntlMixin],
> > +  contextTypes: {
> > +    location: React.PropTypes.object,
> > +    router: React.PropTypes.object,
> > +    intl: React.PropTypes.object
> > +  },
> > +  propTypes: {
> > +    country: React.PropTypes.string.isRequired,
> > +    test: React.PropTypes.string,
> > +    location: React.PropTypes.object
> > +  },
> > +  getInitialState: function() {
> > +    let query = qs.parse(this.props.location.search.replace("?", ""));
>
> Is there a line of code somewhere where we pass in location? It's no
> longer needed and I want to make sure I'm not passing in unused values.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/donate.mozilla.org/pull/1444/files/0e7dc0b6d694a963156e078508b94b06d9bb16a0#r58319204>
>